### PR TITLE
Create a python module and a test for it

### PR DESCRIPTION
**Pull request by m04116**
Issue #3 
Create a python module with dummy function and test for it

### CODE REVIEW

File: test_email_validation.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (webknjaz) [inline test_email_validation.py]: dummy line

File: test_email_validation.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import re
```
Comment (webknjaz) [inline test_email_validation.py]: not needed here

File: test_email_validation.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import unittest
+
+
+class EmailValidationTest(unittest.TestCase):
+    """unit test for EmailValidation"""
+    def test_validate_email(self):
+        self.assertTrue(e_mail, validate_email())
+
+
+def validate_email(a):
```
Comment (webknjaz) [inline test_email_validation.py]: 1) not here
2) argument name sucks
3) should be returning value
4) should have a comprehensive docstring

File: test_email_validation.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    pattern = r'^([a-z]+|[0-9])*@[a-z]+_?[a-z]+\.com$'
```
Comment (webknjaz) [inline test_email_validation.py]: not needed

File: test_email_validation.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    if re.match(pattern, a):
```
Comment (webknjaz) [inline test_email_validation.py]: not needed

File: test_email_validation.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        print('True')
```
Comment (webknjaz) [inline test_email_validation.py]: not needed

File: test_email_validation.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    else:
```
Comment (webknjaz) [inline test_email_validation.py]: not needed

File: test_email_validation.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        print('False')
```
Comment (webknjaz) [inline test_email_validation.py]: not needed

File: test_email_validation.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+
+e_mail = 'vasya@gde-to.ru'
```
Comment (webknjaz) [inline test_email_validation.py]: WTF?

File: test_email_validation.py (hunk 1 above)
Comment (webknjaz) [inline test_email_validation.py]: where's docstring?

File: test_email_validation.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
+
+import re
+import unittest
+
+
+class EmailValidationTest(unittest.TestCase):
+    """unit test for EmailValidation"""
+    def test_validate_email(self):
+        self.assertTrue(e_mail, validate_email())
```
Comment (webknjaz) [inline test_email_validation.py]: Invalid check: see the documentation for the input arguments.

**Review by webknjaz — CHANGES_REQUESTED** [review]
.0. RTFM
.1. Read the task carefully in loop until understood :)

**Comment by webknjaz** [comment]
Make the PR title correspond its contents and add some short description on what you did here

File: .pre-commit-config.yaml [hunk 11]
```diff
@@ -13,7 +13,7 @@
     - id: name-tests-test
     - id: requirements-txt-fixer
 
-# -   repo: git://github.com/FalconSocial/pre-commit-mirrors-pep257
-#     sha: master
-#     hooks:
-#     -   id: pep257
+ -   repo: git://github.com/FalconSocial/pre-commit-mirrors-pep257
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: Indentation style here does not match section above
+

File: email_validation.py [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import re
```
Comment (webknjaz) [inline email_validation.py]: YAGNI

File: email_validation.py [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+
+
+def validate_email(a):
+    pattern = r'^([a-z]+|[0-9])*@[a-z]+_?[a-z]+\.com$'
```
Comment (webknjaz) [inline email_validation.py]: See the task

File: email_validation.py [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+    if re.match(pattern, a):
+        print('True')
+    else:
+        print('False')
+
+
+a = 'vasya@gde-to.ru'
```
Comment (webknjaz) [inline email_validation.py]: rm

File: test_email_validation.py [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+
+
+e_mail = 'vasya@gde-to.ru'
```
Comment (webknjaz) [inline test_email_validation.py]: rm

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: test_email_validation.py [hunk 16]
```diff
@@ -0,0 +1,12 @@
+
+import email_validation
+import unittest
+
+
+class EmailValidationTest(unittest.TestCase):
+    """unit test for EmailValidation"""
+    def test_validate_email(self):
+        self.assertTrue(e_mail, validate_email())
```
Comment (webknjaz) [inline test_email_validation.py]: `validate_email` does not exist in this scope

File: .gitignore [hunk 17]
```diff
@@ -6,6 +6,9 @@ __pycache__/
 # C extensions
 *.so
 
+.virtualenvs/
```
Comment (webknjaz) [inline .gitignore]: normally you don't need to `.gitignore` any of these

File: email_validation.py [hunk 18]
```diff
@@ -0,0 +1,2 @@
+def validate_email(email):
```
Comment (webknjaz) [inline email_validation.py]: correct

File: test_email_validation.py [hunk 19]
```diff
@@ -0,0 +1,8 @@
+import email_validation
```
Comment (webknjaz) [inline test_email_validation.py]: usually stdlib modules are imported first and only after them you import your own project modules

File: test_email_validation.py [hunk 20]
```diff
@@ -0,0 +1,8 @@
...
+import unittest
+
+
+class EmailValidationTest(unittest.TestCase):
+    """unit test for EmailValidation"""
+    def test_validate_email(self):
+        self.assertTrue(email, validate_email())
```
Comment (webknjaz) [inline test_email_validation.py]: 1) This is invalid: `assertTrue()` accepts only one argument for test.
2) `email` variable doesn't exist in this scope

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: test_email_validation.py [hunk 21]
```diff
@@ -5,4 +5,4 @@
 class EmailValidationTest(unittest.TestCase):
     """unit test for EmailValidation"""
     def test_validate_email(self):
-        self.assertTrue(email, validate_email())
+        self.assertTrue('123@123.com', email_validation.validate_email())
```
Comment (webknjaz) [inline test_email_validation.py]: Read the doc

File: test_email_validation.py [hunk 22]
```diff
@@ -0,0 +1,8 @@
+import email_validation
+import unittest
+
+
+class EmailValidationTest(unittest.TestCase):
+    """unit test for EmailValidation"""
```
Comment (webknjaz) [inline test_email_validation.py]: Docstrings are usually sentences (i.e. start with uppercase letter and end with a period)

File: email_validation.py [hunk 23]
```diff
@@ -0,0 +1,6 @@
+"""validate email module."""
```
Comment (webknjaz) [inline email_validation.py]: Uppercase `V`

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: test_email_validation.py (hunk 21 above)
Comment (webknjaz) [reply test_email_validation.py]: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue

**Review by webknjaz — CHANGES_REQUESTED** [review]
Change travis script section to
```
script:
- nosetests
- pre-commit run --all-files
```

File: test_email_validation.py (hunk 19 above)
Comment (webknjaz) [reply test_email_validation.py]: Could you please use `import from` syntax instead?

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: test_email_validation.py [hunk 24]
```diff
@@ -0,0 +1,8 @@
+import email_validation
+import unittest
```
Comment (webknjaz) [inline test_email_validation.py:3]: You may fulfill my previous requirement enabling one of `reorder-python-imports` or `python-import-sorter` in pre-commit's config.

File: .travis.yml [hunk 25]
```diff
@@ -4,6 +4,6 @@ python: 3.5
 cache: pip
 
 before_install: pip install -U nose pre-commit
-script: |
-  nosetests
-  pre-commit run --all-files
+script:
+- nosetests
```
Comment (webknjaz) [inline .travis.yml]: Add `-s -v` arguments to `nosetests`

File: test_email_validation.py (hunk 24 above)
Comment (m04116) [reply test_email_validation.py:3]: "from unittest..." doesn't work. No matches in google(

File: .idea/email-validation.iml [hunk 26]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (webknjaz) [inline .idea/email-validation.iml]: rm

File: .idea/misc.xml [hunk 27]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (webknjaz) [inline .idea/misc.xml]: rm

File: .idea/modules.xml [hunk 28]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (webknjaz) [inline .idea/modules.xml]: rm

File: .idea/vcs.xml [hunk 29]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (webknjaz) [inline .idea/vcs.xml]: rm

File: .idea/workspace.xml [hunk 30]
```diff
@@ -0,0 +1,183 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (webknjaz) [inline .idea/workspace.xml]: rm

File: .travis.yml [hunk 31]
```diff
@@ -4,6 +4,6 @@ python: 3.5
 cache: pip
 
 before_install: pip install -U nose pre-commit
-script: |
-  nosetests
-  pre-commit run --all-files
+script:
+- nosetests -s -v
```
Comment (webknjaz) [inline .travis.yml:8]: Why did you remove running `pre-commit` utility?

File: test_email_validation.py [hunk 32]
```diff
@@ -0,0 +1,8 @@
+from email_validation import validate_email
```
Comment (webknjaz) [inline test_email_validation.py]: You need a module docstring at line 0.

File: test_email_validation.py [hunk 33]
```diff
@@ -0,0 +1,8 @@
...
+import unittest
+
+
+class EmailValidationTest(unittest.TestCase):
+    """Unit test for EmailValidation"""
+    def test_validate_email(self):
+        self.assertTrue(validate_email('123@123.com'))
```
Comment (webknjaz) [inline test_email_validation.py:10]: Please add a docstring to this test method.

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: .idea/email-validation.iml (hunk 26 above)
Comment (m04116) [reply .idea/email-validation.iml]: Have i remove above string?

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
Make sure to refer the issue within your commit message. Like `Close #3`

**Review by webknjaz — CHANGES_REQUESTED** [review]
@m04116
1) re-title this PR to correspond its purpose
2) refer the issue you are solving

**Comment by webknjaz** [comment]
Describe the purpose of this PR.

**Review by webknjaz — DISMISSED** [review]

**Comment by m04116** [comment]
Python module with dummy function created. All tests are passed.

File: email_validation.py [hunk 34]
```diff
@@ -0,0 +1,6 @@
+"""Validate email module."""
```
Comment (webknjaz) [inline email_validation.py]: Re-phrase this into `Email validation module.`

File: email_validation.py [hunk 35]
```diff
@@ -0,0 +1,6 @@
...
+
+
+def validate_email(email):
+    """Email validation."""
```
Comment (webknjaz) [inline email_validation.py:5]: `Validate email against pre-defined rules.`

File: test_email_validation.py [hunk 36]
```diff
@@ -0,0 +1,10 @@
+"""Test module email_validation."""
+from email_validation import validate_email
+import unittest
+
+
+class EmailValidationTest(unittest.TestCase):
+    """Unit test for EmailValidation."""
```
Comment (webknjaz) [inline test_email_validation.py]: test`s`

Comment (webknjaz) [reply test_email_validation.py]: `for email validation module`

File: test_email_validation.py [hunk 37]
```diff
@@ -0,0 +1,10 @@
...
+    def test_validate_email(self):
+        """Test by assertTrue method."""
```
Comment (webknjaz) [inline test_email_validation.py]: `Check whether validate_email function behaves correctly`

**Review by webknjaz — CHANGES_REQUESTED** [review]

**Review by webknjaz — APPROVED** [review]

**Pull request merged**